### PR TITLE
Fix how ShellTask retrieves the Pod class name

### PR DESCRIPTION
**Pull request by matheusMoreno**
# TL;DR
This PR fixes the type check executed at package time of shell tasks with specified task configurations.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description

The `task_config` parameter passed to a shell task must be of type `None` or `flytekitplugins.pod.task.Pod`. Right now, this check is being made with the following logic:

```python
if task_config is not None:
    if str(type(task_config)) != "flytekitplugins.pod.task.Pod":
        raise ValueError("TaskConfig can either be empty - indicating simple container task or a PodConfig.")
```

The nested if condition always returns true, because `str(type(task_config))` returns the string `"<class 'flytekitplugins.pod.task.Pod'>"`.

The proposed solution uses the `__module__` and `__class__.__name__` dunder methods to build the fully qualified class name for the parameter since it appears it shouldn't import `flytekitplugins`.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2765

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by kumare3** [comment]
Ohh good catch ❤️

**Review by kumare3 — APPROVED** [review]

**Comment by wild-endeavor** [comment]
@matheusMoreno could you run `make lint` please? Thank you!

**Comment by matheusMoreno** [comment]
> @matheusMoreno could you run `make lint` please? Thank you!

Done!

**Review by pingsutw — APPROVED** [review]
